### PR TITLE
Add high-order gmsh elements

### PR DESCRIPTION
**Pull request by alexfikl**
This adds all the complete high-order elements supported by gmsh (for triangles, tets, quads and hexes at least).

The triangle and tet node tuples all match for the orders that were already there. 

The quads and hexes are not the same though and I'm not sure if that changed in some gmsh version or was always off. The new node tuples fix the hex mesh generation I was seeing in meshmode [here](https://github.com/inducer/meshmode/pull/70/files#diff-e46b9dad1281f041c12625824e53386f096319c7bb20fea9dcd48ed4f4743c92R1315).

### CODE REVIEW

**Comment by alexfikl** [comment]
Also, inducer/meshmode#70 is currently running a CI with this branch, so that should hopefully scream if this managed to change the order in weird ways.

File: contrib/gmsh-node-tuples.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        "MSH_HEX_8": 5,
+        "MSH_HEX_27": 12,
+        "MSH_HEX_64": 92,
+        "MSH_HEX_125": 93,
+        "MSH_HEX_216": 94,
+        "MSH_HEX_343": 95,
+        "MSH_HEX_512": 96,
+        "MSH_HEX_729": 97,
+        "MSH_HEX_1000": 98,
+        }
```
Comment (inducer) [inline contrib/gmsh-node-tuples.py:70]: It looks like this information (the element type IDs) is maintained redundantly here and in `gmsh_interop/reader.py`. Could you add that to the generated bits so we don't have to have it twice?

**Review by inducer — COMMENTED** [review]
Thanks! This is much nicer than the crappy process by which I obtained the original node tuples. LGTM aside from my one comment.

File: contrib/gmsh-node-tuples.py (hunk 1 above)
Comment (alexfikl) [reply contrib/gmsh-node-tuples.py:70]: Fixed in 7e53562

Added an `element_type` to each class and used that to generate the `gmsh_element_type_to_info_map`.

**Comment by inducer** [comment]
Thanks!

**Pull request merged**